tests/integration/backend.go

```diff
@@ -0,0 +1,60 @@
+package integration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/ethclient/simulated"
+	"github.com/stretchr/testify/require"
+)
+
+type Backend interface {
+	simulated.Client
+	Commit() common.Hash
+}
+
+// --- geth simulated backend ---
+
+type simulatedBackendWrapper struct {
+	backend *simulated.Backend
+	simulated.Client
+}
+
+func (w *simulatedBackendWrapper) Commit() common.Hash {
+	return w.backend.Commit()
+}
+
+func NewSimulatedBackend(t *testing.T, genesisAlloc types.GenesisAlloc) Backend {
+	t.Helper()
+
+	gethBackend := simulated.NewBackend(genesisAlloc)
+
+	return &simulatedBackendWrapper{
+		backend: gethBackend,
+		Client:  gethBackend.Client(),
+	}
+}
+
+// --- rpc backend ---
+
+type rpcBackend struct {
+	simulated.Client
+}
+
+func (b *rpcBackend) Commit() common.Hash {
```

Comment:
the `Commit` method is not really needed for the rpc backend; the geth service will mine a few block every X seconds.